File: 3.cpp
#include <iostream>
#include <vector>

int maxMoneyLeft() {
    int n, m;
    std::cin >> n >> m;
    std::vector <int> gifts(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> gifts[i];
    }
    int currentmax = m;
    for (int i = 0; i < n; i++) {
        if (currentmax >= gifts[i]) {
            int leftMax = gifts[i] - 1;
            int rightMax = currentmax - gifts[i];
            currentmax = std::max(leftMax, rightMax);
        }
    }
    return currentmax;
}

int main()
{
    std::cout << maxMoneyLeft() << std::endl;
    return 0;
}

File: 3.js
function maxMoneyLeft (credLim, gifts) {
    const resultarr = [];
    for (let index = 1; index <= credLim; index++) {
        let credit = index;
        for(let gift of gifts) {
            if(credit >= gift) {
                credit -= gift;
            } 
        }
        resultarr.push(credit);
    }
    return Math.max(...resultarr);
}

const credLim = 10;
const gifts = [5,4,1];
const n = 3;
